### PR TITLE
Implemented developers token generation. Resolves #12

### LINKED ISSUE #12: Development tokens for SDK
Hi,

It would be great to add development tokens into the Stream SDK.

### PR DESCRIPTION
**Pull request by sierpinskid**
* Implemented dev tokens
* Fixed error printing appending to previous error
* Added joining a member to a channel after creating it
* Fixed serializer peek method for entry being JObject case
* Added better error printing when APIError is returned from an API call

### CODE REVIEW

File: Assets/Plugins/StreamChat/Core/Models/APIError.cs [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    {
+        /// <summary>
+        /// Response HTTP status code
+        /// </summary>
+        public double? StatusCode { get; set; }
+
+        /// <summary>
+        /// API error code
+        /// </summary>
+        public double? Code { get; set; }
```
Comment (peterdeme) [inline Assets/Plugins/StreamChat/Core/Models/APIError.cs]: int maybe?

File: Assets/Plugins/StreamChat/Core/Models/APIError.cs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+        /// <summary>
+        /// Request duration
+        /// </summary>
+        public string Duration { get; set; }
+
+        /// <summary>
+        /// Additional error info
+        /// </summary>
+        public System.Collections.Generic.IDictionary<string, string> ExceptionFields { get; set; }
```
Comment (peterdeme) [inline Assets/Plugins/StreamChat/Core/Models/APIError.cs:28]: Why fully qualified name? Idictionary already defined somewhere else? :)

File: Assets/Plugins/StreamChat/Core/StreamChatClient.cs [hunk 3]
```diff
@@ -74,6 +77,21 @@ public static IStreamChatClient CreateDefaultClient(AuthCredentials authCredenti
...
         }
 
+        /// <summary>
+        /// Create Development Authorization Token. Dev tokens work only if you enable "Disable Auth Checks" in your project's Dashboard.
+        /// Dev tokens bypasses authorization and should only be used during development and never in production!
+        /// More info <see cref="https://getstream.io/chat/docs/unity/tokens_and_authentication/?language=unity#developer-tokens"/>
+        /// </summary>
+        public static string CreateDeveloperAuthToken(string userId)
+        {
+            var header = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9"; //  header content = {"alg": "HS256", "typ": "JWT"}
```
Comment (peterdeme) [inline Assets/Plugins/StreamChat/Core/StreamChatClient.cs:87]: Nice optimization :)

**Review by peterdeme — APPROVED** [review]
lgtm, 2 questions

File: Assets/Plugins/StreamChat/Core/Models/APIError.cs (hunk 2 above)
Comment (sierpinskid) [reply Assets/Plugins/StreamChat/Core/Models/APIError.cs:28]: no 😄 I create all models from DTOs which are auto-generated by the OpenApi generator, and there isn't much point in clearing those. I'll create out the whole structure in a separate PR

File: Assets/Plugins/StreamChat/Core/StreamChatClient.cs (hunk 3 above)
Comment (sierpinskid) [reply Assets/Plugins/StreamChat/Core/StreamChatClient.cs:87]: took it from android SDK 😛

File: Assets/Plugins/StreamChat/Core/Models/APIError.cs (hunk 2 above)
Comment (peterdeme) [reply Assets/Plugins/StreamChat/Core/Models/APIError.cs:28]: oh, no need to. generated code can stay as it is :)

**Pull request merged**